src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -658,12 +786,15 @@ private void ParseTargetElement(Target target, ILoggingConfigurationElement targ
             }
         }
 
-        private bool ParseTargetWrapper(Dictionary<string, ILoggingConfigurationElement> typeNameToDefaultTargetParameters, string name, ILoggingConfigurationElement childElement,
-    WrapperTargetBase wrapper)
+        private bool ParseTargetWrapper(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)